sqlite3/lib/src/impl/function_store.dart

```diff
@@ -71,9 +87,32 @@ class FunctionPointerAndData {
     this.xFunc,
     this.xStep,
     this.xFinal,
+    this.xCompare,
   });
 }
 
+int _collatingFunctionImpl(
+  Pointer<Void> udp,
+  int sizeA,
+  Pointer<Void> textA,
+  int sizeB,
+  Pointer<Void> textB,
+) {
+  final functionId = udp.address;
+  final target = functionStore.readCollating(functionId);
+
+  final Uint8List bytesA = textA.copyRange(sizeA);
+  final Uint8List bytesB = textB.copyRange(sizeB);
+  final String txtA = String.fromCharCodes(bytesA);
+  final String txtB = String.fromCharCodes(bytesB);
```

Comment:
I'm fairly sure `fromCharCodes` copies internally, so we don't need to copy the pointers into Dart's heap first.

Can we just use something like `String.fromCharCodes(textA.cast<Uint8>().asTypedList(sizeA))` and avoid the `copyRange` call?

Reply:
And actually, my understanding is that `textA` and `textB` contain the text as a utf8 bytes? In that case we can just use `textA.cast<sqlite3_char>().readString(sizeA)`. `String.fromCharCodes` would only be correct if the texts were encoded with utf16-BE (I think?).

Maybe we should add a text using higher unicode chars like emoji to make sure we're getting the encoding right here.

Reply:
It's pretty new for us to play with String by this way, we will change that.

``textA`` and ``textB`` contain text encoded in the encoding specified by the ``eTextRep`` argument.

As **sqlite3.dart** use only UTF8 for the moment, may we can simply keep reading the text as UTF8?

Reply:
Yup, reading as utf-8 sounds like the safest idea to me.